tests/base/utilities_pack_unpack_03.cc

```diff
@@ -0,0 +1,77 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// test Utilities::pack/unpack on some types. this test checks that
+// for trivially-copyable (small) types, packing is just a memcpy
+// operation
+
+#include "../tests.h"
+
+#include <deal.II/base/utilities.h>
+#include <deal.II/base/point.h>
+#include <boost/serialization/utility.hpp>
+
+#include <tuple>
+#include <cstring>
+
+struct X
+{
+  int i;
+  int k;
+  double d;
+
+  bool operator == (const X &x) const
+  {
+    return i==x.i && k==x.k && d==x.d;
+  }
+
+  template <class Archive>
+  void serialize(Archive &ar, const unsigned int)
+  {
+    ar   &i &k &d;
+  }
+};
+
+
+
+template <typename T>
+void check (const T &object)
+{
+  const std::vector<char> buffer = Utilities::pack (object);
+  if (!(buffer.size() == sizeof(object))
+      ||
+      !(std::memcmp(buffer.data(), &object, buffer.size()) == 0)
```

Comment:
Can you elaborate on this check? Doesn't this mean that you expect `buffer.data()` to return a `nullptr`?

Reply:
No -- `std::memcmp` compares the data pointed to by the first two arguments (both pointers) with length given by the last argument and returns zero if the two memory locations have identical content.

Reply:
Ah, I see. I confused it with `std::memcpy`.

Reply:
I had to stare at this if clause for quite a while to understand it if that makes you feel better. ;-)

Reply:
I would have written them as separate AssertThrows.